config/reports/helpers.js

```diff
@@ -674,36 +674,51 @@ export function enrichValidationReport(
 }
 
 export async function saveDatasetToNamedGraphs(dataset, namedGraphs) {
-  const writer = new Writer({ format: "N-Triples" });
+  let batch = [];
+  const insertBatch = async () => {
+    const ttl = await quadsToTtl(batch);
+    await updateSudo(`
+        PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+
+        INSERT {
+            GRAPH ?g {
+                ${ttl}
+            }
+        } WHERE {
+          VALUES ?g {
+            ${namedGraphs.map((g) => sparqlEscapeUri(g)).join("\n")}
+          }
+          ?g ext:ownedBy ?someone .
+        }`);
+  };
+  const limit = 1000;
   for (const quad of dataset) {
-    writer.addQuad(quad);
+    batch.push(quad);
+    if (batch.length >= limit) {
+      await insertBatch();
+      batch = [];
+    }
+  }
+  if (batch.length > 0) {
+    await insertBatch();
   }
+}
 
-  const triples = await new Promise((resolve, reject) => {
+export async function quadsToTtl(quads) {
+  const result = new Promise((resolve, reject) => {
+    const writer = new Writer({ format: "N-Triples" });
+    quads.forEach((quad) => {
+      writer.addQuad(quad);
+    });
```

Comment:
Should be possible to just do this
```suggestion
writer.addQuads(quads);
```
ref: https://github.com/lblod/form-content-service/blob/5c74699e9923b646c228e582d19a16c16043a1ef/form-validator.ts#L205